src/data.jl

```diff
@@ -394,6 +493,63 @@ function get_discretized_R_gates(gate_type::String, R::Dict{String, Any}, discre
     return R 
 end
 
+function get_all_CU_gates(params::Dict{String, Any}, elementary_gates::Array{String,1})
+
+    U_gates_ids = findall(x -> startswith(x, "CU"), elementary_gates)
+
+    U_complex = Dict{String, Any}()
+
+    if length(U_gates_ids) >= 1 
+        for i=1:length(U_gates_ids)
+            gate_name = elementary_gates[U_gates_ids[i]]
+            if startswith(gate_name, "CU3")        
+                U_complex[gate_name] = Dict{String, Any}()    
+                
+                for angle in ["θ","ϕ","λ"]
+                    if isempty(params["CU_$(angle)_discretization"])
+                        Memento.error(_LOGGER, "Empty $(angle) discretization angle for U3 gate. Input at least one angle")
+                    end
+                end
+
+                U_complex[gate_name] = QCO.get_discretized_CU3_gates(gate_name, U_complex[gate_name], collect(float(params["CU_θ_discretization"])), collect(float(params["CU_ϕ_discretization"])), collect(float(params["CU_λ_discretization"])), params["num_qubits"])
+            end
+
+            # Add support for U1 and U2 universal gates here
```

Comment:
@lockwo You mean CU1 and CU2?

Reply:
Yea

Reply:
Please update it, though not very important!

Reply:
Ok updated